File: scripts/lookup_comcigan.js
const http = require('http');
const iconv = require('iconv-lite');

const schoolName = process.argv[2] || '양동';

console.log(`🔍 [Comcigan CLI] 컴시간 서버 접속 및 학교 검색 중: "${schoolName}"...`);

const landingUrl = 'http://xn--s39aj90b0nb2xw6xh.kr';

function fetchUrl(url) {
  return new Promise((resolve, reject) => {
    http.get(url, (res) => {
      let chunks = [];
      res.on('data', chunk => chunks.push(chunk));
      res.on('end', () => resolve(Buffer.concat(chunks)));
    }).on('error', reject);
  });
}

function encodeCP949Hex(str) {
  const buf = iconv.encode(str, 'euc-kr');
  return Array.from(buf).map(b => '%' + b.toString(16).toLowerCase().padStart(2, '0')).join('');
}

(async () => {
  try {
    const landingBuf = await fetchUrl(landingUrl);
    const landingHtml = landingBuf.toString('utf-8');

    const frameMatch = landingHtml.match(/<frame\s+[^>]*src=["']([^"']+)["']/i);
    if (!frameMatch) throw new Error('컴시간 프레임 URL추출 실패');

    const frameUrl = new URL(frameMatch[1], landingUrl).href;
    const baseUrl = `${new URL(frameUrl).protocol}//${new URL(frameUrl).host}`;

    const frameBuf = await fetchUrl(frameUrl);
    const frameHtml = frameBuf.toString('utf-8');

    const searchPathMatch = frameHtml.match(/url\s*:\s*'\s*\.\/([^']+)'/i);
    if (!searchPathMatch) throw new Error('컴시간 검색 경로 추출 실패');

    const extractCode = '/' + searchPathMatch[1];
    const hexQuery = encodeCP949Hex(schoolName);
    const searchUrl = baseUrl + extractCode + hexQuery;

    const resultBuf = await fetchUrl(searchUrl);
    let resultText = resultBuf.toString('utf-8');

    const cleanJson = resultText.substring(0, resultText.lastIndexOf('}') + 1);
    const data = JSON.parse(cleanJson);
    const list = data['학교검색'] || [];

    console.log(`\n✅ [Comcigan CLI] 검색 성공! (서버: ${baseUrl})`);
    console.log(`총 ${list.length}건 검색 결과:`);
    console.log('==================================================');
    list.forEach((item, idx) => {
      console.log(`${idx + 1}. [지역: ${item[1]}] ${item[2]} ➔ 학교코드: \x1b[36m${item[3]}\x1b[0m`);
    });
    console.log('==================================================\n');
  } catch (e) {
    console.error('❌ 검색 중 오류 발생:', e.message);
  }
})();

File: scripts/lookup_comcigan.py
import sys
import json
import urllib.request
import urllib.parse
import re

school_name = sys.argv[1] if len(sys.argv) > 1 else "양동중"

print(f'🔍 [Comcigan Python CLI] 컴시간 서버 접속 및 학교 검색 중: "{school_name}"...')

landing_url = 'http://xn--s39aj90b0nb2xw6xh.kr'

try:
    # 1. Fetch Landing Page
    req = urllib.request.Request(landing_url, headers={'User-Agent': 'Mozilla/5.0'})
    with urllib.request.urlopen(req) as resp:
        landing_html = resp.read().decode('utf-8', errors='ignore')

    frame_match = re.search(r'''<frame\s+[^>]*src=["']([^"']+)["']''', landingHtml if 'landingHtml' in locals() else landing_html, re.I)
    if not frame_match:
        raise Exception("컴시간 프레임 URL 추출 실패")

    frame_url = urllib.parse.urljoin(landing_url, frame_match.group(1))
    parsed_frame = urllib.parse.urlparse(frame_url)
    base_url = f"{parsed_frame.scheme}://{parsed_frame.netloc}"

    # 2. Fetch Frame HTML
    req_frame = urllib.request.Request(frame_url, headers={'User-Agent': 'Mozilla/5.0'})
    with urllib.request.urlopen(req_frame) as resp_frame:
        frame_html = resp_frame.read().decode('utf-8', errors='ignore')

    search_path_match = re.search(r'''url\s*:\s*'\s*\.\/([^']+)' ''', frame_html, re.I)
    if not search_path_match:
        raise Exception("컴시간 검색 경로 추출 실패")

    extract_code = '/' + search_path_match.group(1).strip()

    # 3. CP949 URL Encode query
    cp949_bytes = school_name.encode('euc-kr', errors='ignore')
    hex_query = ''.join([f"%{b:02x}" for b in cp949_bytes])

    search_url = base_url + extract_code + hex_query

    # 4. Query Comcigan Search Endpoint
    req_search = urllib.request.Request(search_url, headers={'User-Agent': 'Mozilla/5.0'})
    with urllib.request.urlopen(req_search) as resp_search:
        result_text = resp_search.read().decode('utf-8', errors='ignore')

    clean_json = result_text[:result_text.rfind('}') + 1]
    data = json.loads(clean_json)
    school_list = data.get('학교검색', [])

    print(f"\n✅ [Comcigan Python CLI] 검색 성공! (서버: {base_url})")
    print(f"총 {len(school_list)}건 검색 결과:")
    print("==================================================")
    for idx, item in enumerate(school_list, 1):
        print(f"{idx}. [지역: {item[1]}] {item[2]} ➔ 학교코드: \033[36m{item[3]}\033[0m")
    print("==================================================\n")

except Exception as e:
    print(f"❌ 검색 오류 발생: {e}")
